### ISSUE TITLE
KeyError: 'commit' when loading data from GitHub repository using branch name

### ISSUE THREAD

**Issue by xhiroga** [jerryjliu/llama_index#1194]
## Minimum reproduction code

```python
import os

from llama_index import GPTSimpleVectorIndex, GithubRepositoryReader

loader = GithubRepositoryReader(
    github_token=os.environ["GITHUB_TOKEN"],
    owner="jerryjliu",
    repo="llama_index",
    use_parser=False,
    verbose=True,
    ignore_directories=["examples"],
)

docs = loader.load_data(branch='main')

for doc in docs:
    print(doc.extra_info)

index = GPTSimpleVectorIndex.from_documents(docs)
index.save_to_disk('data/llama_index.index.json')

```

## Error

```error
% make data/llamaindex_index.json
.venv/bin/dotenv run -- .venv/bin/python scripts/create_index_llamaindex.py
Traceback (most recent call last):
  File "/Users/hiroga/.ghq/github.com/xhiroga/agent/scripts/create_index_llamaindex.py", line 13, in <module>
    docs = loader.load_data(branch="main")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/hiroga/.ghq/github.com/xhiroga/agent/.venv/lib/python3.11/site-packages/llama_index/readers/github_readers/github_repository_reader.py", line 185, in load_data
    return self._load_data_from_branch(branch)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/hiroga/.ghq/github.com/xhiroga/agent/.venv/lib/python3.11/site-packages/llama_index/readers/github_readers/github_repository_reader.py", line 147, in _load_data_from_branch
    branch_data: GitBranchResponseModel = self._loop.run_until_complete(
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/hiroga/.ghq/github.com/xhiroga/agent/.venv/lib/python3.11/site-packages/llama_index/readers/github_readers/github_api_client.py", line 283, in get_branch
    return GitBranchResponseModel.from_json(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/hiroga/.ghq/github.com/xhiroga/agent/.venv/lib/python3.11/site-packages/dataclasses_json/api.py", line 65, in from_json
    return cls.from_dict(kvs, infer_missing=infer_missing)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/hiroga/.ghq/github.com/xhiroga/agent/.venv/lib/python3.11/site-packages/dataclasses_json/api.py", line 72, in from_dict
    return _decode_dataclass(cls, kvs, infer_missing)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/hiroga/.ghq/github.com/xhiroga/agent/.venv/lib/python3.11/site-packages/dataclasses_json/core.py", line 159, in _decode_dataclass
    field_value = kvs[field.name]
                  ~~~^^^^^^^^^^^^
KeyError: 'commit'
make: *** [data/llamaindex_index.json] Error 1
```

## Note

If use `docs = loader.load_data('703073ef41f0dd035cdf5c720d9e8b53eaf18bbd')`, the same error does not happen.

## Environment
```toml
[tool.poetry.dependencies]
python = "^3.11.2"
slack-bolt = ">=1.17.0,<2"
openai = ">=0.27,<0.28"
tiktoken = ">=0.3.0,<0.4"
langchain = "^0.0.138"
llama-index = "^0.5.13.post1"
```

**Comment by cosmingamanusi** [comment]
I have the same issue too 👍

**Comment by dciborow** [comment]
seeing the same issue, also using Python 3.11 if that helps.

**Comment by xhiroga** [comment]
I have investigated this issue. Although I couldn't reproduce the original issue, I found another error.

In httpx, when the response contains an error, it is necessary to explicitly raise the error.

https://user-images.githubusercontent.com/13391129/233757494-a483afa1-0e35-441f-aa3f-3c817135dd39.mov

(Reference: <https://www.python-httpx.org/exceptions/#exception-classes> )

This was not being handled, which caused the error to be difficult to understand. I plan to fix this issue.
